components/Themes/index.js

```diff
@@ -40,50 +42,83 @@ const ThemeItem = ({ children, item, isSelected, onClick }) => (
 
 const themeIcon = <ThemeIcon />
 
+const getCustomName = themes =>
+  `Custom Theme ${themes.filter(({ name }) => name.startsWith('Custom Theme')).length + 1}`
+
 class Themes extends React.PureComponent {
   selectedTheme = DEFAULT_THEME
 
   state = {
     themes: THEMES,
     preset: this.props.theme,
-    highlights: {},
     name: 'Custom Theme',
     selected: null
   }
 
   dropdown = React.createRef()
 
   componentDidMount() {
+    const { queryState } = getRouteState(this.props.router)
+
+    const queryHighlights = queryState
```

Comment:
Since `highlights` are just passed in as a prop now, shouldn't `queryHighlights` be calculated in `Editor` and passed in directly to `<Themes />`

Reply:
@mixn is passing a single `highlights` query param as JSON sufficient for the CLI? 🙂

Reply:
@mfix22 Could you provide an example of usage via URL (in the deployment below)? That’d be great. 😊

Reply:
@mfix22 Tried [this](https://carbon-ek0fn70mc.now.sh/?bg=rgba(171%2C%20184%2C%20195%2C%201)&t=theme%3A45t3iqdbm4b&wt=none&l=auto&ds=true&dsyoff=20px&dsblur=68px&wc=true&wa=true&pv=56px&ph=56px&ln=false&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false) — unfortunately didn’t work for me? Sorry if I’m missing something.

Reply:
@mixn it's expecting the highlight url param to equal something of the form `encodeURIComponent(JSON.stringify({ background: '#fff', text: '#000', ... }))` 

Have you tried something like that?

Reply:
@mixn it should work with a queryString of the form: `?state=${btoa(encodeURIComponent(JSON.stringify({ highlights: { background: 'blue', operator: '#000', text: '#fff', ... } })))}`

Reply:
Hey @mixn yeah it's not ready quite yet, I was more asking from an API standpoint. I'll let you know soon.

Reply:
Hey @mixn yeah it's not ready quite yet, I was more asking from an API standpoint. I'll let you know soon.

Reply:
Hey @mixn yeah it's not ready quite yet, I was more asking from an API standpoint. I'll let you know soon.

Reply:
@mixn Thanks for letting me know. :) It definitely sounds sufficient. Would be amazing if it’d be optional as the others and would behave the same way. I prepare the URL [here](https://github.com/mixn/carbon-now-cli/blob/master/cli.js#L193) after it has gone through all previous steps, it’s really, really basic. Initially it’s based on [the default settings](https://github.com/mixn/carbon-now-cli/blob/master/src/helpers/default-settings.js).

The only small problem I see is it adding to [this issue](https://github.com/mixn/carbon-now-cli/issues/44) / the length of URLs. But that’s a separate thing. :)